docs/source/configuration/traffic-shaping.mdx

```diff
@@ -44,7 +33,135 @@ traffic_shaping:
+
+## Subgraph side traffic shaping
+
+The Apollo Router supports various options affecting traffic with subgraphs, that can either be defined for all subgraphs,
+or overriden per subgraph:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    deduplicate_query: true # Enable query deduplication for all subgraphs.
+  subgraphs: # Rules applied to requests from the router to individual subgraphs
+    products:
+      deduplicate_query: false # Disable query deduplication for the products subgraph.
+```
+
+### Compression
+
+The Apollo Router can compress request bodies to subgraphs (along with response bodies to clients).
+It currently supports these algorithms: `gzip`, `br`, and `deflate`.
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    compression: br # Enable brotli compression for all subgraphs.
+```
+
+### Rate limiting
+
+Subgraph request rate limiting uses the same configuration as client rate limiting, and is calculated
+per subgraph, not per backend host.
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    global_rate_limit: # Accept a maximum of 10 requests per 5 secs. Excess requests must be rejected.
+      capacity: 10
+      interval: 5s # Must not be greater than 18_446_744_073_709_551_615 milliseconds and not less than 0 milliseconds
+```
+
+### Experimental request retry
 
+On failure, subgraph requests can be retried automatically. This is deactivated by default for mutations.
+This uses [Finagle's *RetryBudget* algorithm](https://finagle.github.io/blog/2016/02/08/retry-budgets/),
+in which every successful request adds an expirable token to a bucket, and every retry consumes a number
+of those tokens. On top of that, a minimal number of retries per second is available, to test regularly
+when the retry budget was entirely consumed or on startup when very few requests have been sent.
+The tokens expire so the budget has a large number of available retries if a lot of recent requests were
+successful, but reduces quickly on frequent failures to avoid sending too much traffic to the subgraph.
+
+It is configurable as follows:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    experimental_retry:
+      min_per_sec: 10 # minimal number of retries per second (`min_per_sec`, default is 10 retries per second)
+      ttl: 10s # for each successful request, we register a token, that expires according to this option (default: 10s)
+      retry_percent: 0.2 # defines the proportion of available retries to the current number of tokens
+      retry_mutations: false # allows retries on mutations. This should only be enabled if mutations are idempotent
 ```
 
-Any configuration under the `subgraphs` key takes precedence over configuration under the `all` key. In the example above, query deduplication is enabled for all subgraphs _except_ the `products` subgraph.
+### Variable deduplication
```

Comment:
Not this PR, but...

Why is this configurable? Why wouldn't you always want to do this? Same for Query deduplication...

I vaguely remember there was a reason for this at one time, but I can't remember the reason and I wonder if it's no longer relevant.

Reply:
IIRC The reason was mainly that we don't know if it's stable enough, so it should have an option to deactivate it. I agree that it should always be active

Reply:
I think variable de-duplication is probably fine.

> Query deduplication

This does tend to have more side effects (behind the scenes things), generally speaking, which is why we initially avoided it.

Reply:
I opened https://github.com/apollographql/router/issues/2387 to keep track of that